src/controllers/AuthController.js

```diff
@@ -0,0 +1,94 @@
+import { hashSync, genSaltSync } from 'bcryptjs';
+import Mailer from '../utils/mailer';
+import db from '../models';
+import Helpers from '../utils/helpers';
+import { ApiResponse } from '../utils/index';
+
+
+const { generateToken, verifyToken } = Helpers;
+const { User } = db;
+const { sendResetMail } = Mailer;
+
+/**
+ * Collection of methods for Auth
+ * @class AuthController
+ */
+class Auth {
+  /**
+   * Sends a user reset password link
+   *
+   * @param {Request} req - The request from the endpoint.
+   * @param {Response} res - The response returned by the method.
+   * @returns {JSON} A JSON response with a successfully message.
+   * @memberof Auth
+   */
+  static async sendResetPasswordEmail(req, res) {
+    try {
+      const { email } = req.body;
+      const user = await User.findOne({ where: { email } });
+      if (!user) {
+        return res.status(404).json(new ApiResponse(false, 404, 'User account does not exist'));
```

Comment:
Throw an error here. Could be `new ApiError(404, User account does not exist)`, it will be caught in the catch block and sent to the user
This could lead to a bug where a response is sent to the user after it has already been sent once because the code did not stop executing after the first response has been sent. Throwing an error instead moves code execution straight to the catch block
```suggestion
        throw ApiError(404, 'User account does not exist');
```

Reply:
The "return" keyword essentially terminates the execution of a function call returning the value assigned to the return call itself. With the return keyword specified, no further responses would be sent to the client when the **if** condition in the method scope evaluates to true. So I think his implementation isn't wrong, except we want to introduce some conventions to follow as team.

Reply:
Exactly my reason behind the implementation @daylay92. @Segun-Ogundipe what do you think?